src/models/event.js

```diff
@@ -299,6 +299,58 @@ utils.extend(module.exports.MatrixEvent.prototype, {
     },
 
     /**
+     * Attempt to decrypt this event.
+     *
+     * (This is used within the SDK: it isn't intended for use by applications)
+     *
+     * @internal
+     *
+     * @param {module:crypto} crypto crypto module
+     */
+    attemptDecryption: function(crypto) {
+        if (!crypto) {
+            this._badEncryptedMessage("Encryption not enabled");
+            return;
+        }
+
+        if (!this.isEncrypted()) {
+            throw new Error("Attempt to decrypt event which isn't encrypted");
+        }
+
+        if (
+            this._clearEvent && this._clearEvent.content &&
+                this._clearEvent.content.msgtype !== "m.bad.encrypted"
+        ) {
+            // we may want to just ignore this? let's start with rejecting it.
+            throw new Error(
+                "Attempt to decrypt event which has already been encrypted",
+            );
+        }
+
+        try {
+            crypto.decryptEvent(this);
+        } catch (e) {
```

Comment:
Is this actually ever going to do anything? The previous commit changed `decryptEvent` to be:
```js
OlmDecryption.prototype.decryptEvent = async function(event) {
```
Meaning this function will never throw, it will always return a Promise. If you `yield`ed then it might throw though, but you don't.